lib/rules/id-length.js

```diff
@@ -28,7 +30,7 @@ module.exports = function(context) {
                 // regular property assignment
                 parent.parent.left === parent || (
                     // or the last identifier in an ObjectPattern destructuring
-                    parent.parent.type === "Property" && parent.parent.value === parent &&
+                    parent.parent.type === "Property" && properties && parent.parent.value === parent &&
```

Comment:
Correct. I find these destructuring thing a bit confusing :( Anyways yeah, thanks for giving it a shot!